### ISSUE TITLE
Delete hook, endless cycle

### ISSUE THREAD

**Issue by lionskape** [flant/shell-operator#82]
After resolving some issues (e.g. broken hook names) I have installed shell operator as Deployment app.

I used the configuration "secret-copier" from examples, but it goes into an endless loop.
Can you help mew with that?

The log is listed below.

```
2019-12-02T01:34:00Z INFO     : shell-operator master-a10d5213a3fea52c3f426502b79d8cd60adc28ae
2019-12-02T01:34:00Z INFO     : HTTP SERVER Listening on 0.0.0.0:9115
2019-12-02T01:34:00Z INFO     : Working dir: /hooks
2019-12-02T01:34:00Z INFO     : Use temporary dir: /tmp/shell-operator
2019-12-02T01:34:00Z INFO     : Initialize hooks manager. Search for and load all hooks.
2019-12-02T01:34:00Z INFO     : Load hook config from '/hooks/add_or_update_secret'
2019-12-02T01:34:00Z INFO     : Load hook config from '/hooks/create_namespace'
2019-12-02T01:34:00Z INFO     : Load hook config from '/hooks/delete_secret'
2019-12-02T01:34:00Z INFO     : Load hook config from '/hooks/schedule_sync_secret'
2019-12-02T01:34:00Z INFO     : Initializing schedule manager ...
2019-12-02T01:34:00Z INFO     : KUBE Init Kubernetes client
2019-12-02T01:34:00Z INFO     : KUBE-INIT Kubernetes client is configured successfully
2019-12-02T01:34:00Z INFO     : MAIN: run main loop
2019-12-02T01:34:00Z INFO     : QUEUE add all HookRun@OnStartup
2019-12-02T01:34:00Z INFO     : MSTOR Create new metric shell_operator_live_ticks
2019-12-02T01:34:00Z INFO     : MSTOR Create new metric shell_operator_tasks_queue_length
2019-12-02T01:34:00Z INFO     : Start monitors: 3
2019-12-02T01:34:00Z INFO     : Running schedule manager ...
2019-12-02T01:34:00Z INFO     : Resync Event
2019-12-02T01:34:00Z INFO     : Resync Event
2019-12-02T01:34:00Z INFO     : Resync Event
2019-12-02T01:34:00Z INFO     : EVENT Kube event '14adeee3-c4b6-4c4b-b32a-077c68a8c0ea'
2019-12-02T01:34:00Z INFO     : QUEUE add TASK_HOOK_RUN@ON_KUBERNETES_EVENT add_or_update_secret
2019-12-02T01:34:00Z INFO     : EVENT Kube event '26ce9088-d44f-4d9e-86d3-692e3192292a'
2019-12-02T01:34:00Z INFO     : QUEUE add TASK_HOOK_RUN@ON_KUBERNETES_EVENT delete_secret
2019-12-02T01:34:00Z INFO     : EVENT Kube event 'ece3742b-43b9-442b-ad10-b6fffd0ff97b'
2019-12-02T01:34:00Z INFO     : QUEUE add TASK_HOOK_RUN@ON_KUBERNETES_EVENT create_namespace
2019-12-02T01:34:03Z INFO     : TASK_RUN HookRun@ON_KUBERNETES_EVENT add_or_update_secret
2019-12-02T01:34:03Z INFO     : Running hook 'add_or_update_secret' binding 'ON_KUBERNETES_EVENT' ...
Got Synchronization event
2019-12-02T01:34:03Z INFO     : TASK_RUN HookRun@ON_KUBERNETES_EVENT delete_secret
2019-12-02T01:34:03Z INFO     : Running hook 'delete_secret' binding 'ON_KUBERNETES_EVENT' ...
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
2019-12-02T01:34:05Z INFO     : MSTOR Create new metric shell_operator_hook_errors
2019-12-02T01:34:05Z ERROR    : TASK_RUN TASK_HOOK_RUN 'delete_secret' on 'ON_KUBERNETES_EVENT' failed. Will retry after delay. Failed count is 1. Error: delete_secret FAILED: exit status 1
2019-12-02T01:34:05Z INFO     : TASK_RUN Delay for 5s
2019-12-02T01:34:10Z INFO     : TASK_RUN HookRun@ON_KUBERNETES_EVENT delete_secret
2019-12-02T01:34:10Z INFO     : Running hook 'delete_secret' binding 'ON_KUBERNETES_EVENT' ...
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
Error from server (NotFound): secrets "null" not found
```

**Comment by diafour** [comment]
It's a problem with jq expressions in an example. Hooks are using a new config, but jq expressions are for old binding context. I will fix this soon. Thank you for pointing this out, I need a closer look for examples.

**Comment by diafour** [comment]
Fixed in #83.
